src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -490,6 +494,41 @@ void RegenerateDatabaseIdInIndexes(RavenConfiguration configuration, DocumentDat
             }
         }
 
+        private static void RemoveSubscriptionFromDatabaseValues(RestoreSettings restoreSettings)
+        {
+            foreach (var keyValue in restoreSettings.DatabaseValues)
+            {
+                if (keyValue.Key.StartsWith(SubscriptionState.Prefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    var subscriptionState = JsonDeserializationClient.SubscriptionState(keyValue.Value);
+                    restoreSettings.Subscriptions.Add(keyValue.Key, subscriptionState);
+                }
+            }
+
+            foreach (var keyValue in restoreSettings.Subscriptions)
+            {
+                restoreSettings.DatabaseValues.Remove(keyValue.Key);
+            }
+        }
+
+        private static async Task HandleSubscriptionFromSnapshot(List<string> filesToRestore, Dictionary<string, SubscriptionState> subscription, 
+            string databaseName, DocumentDatabase database)
+        {
+            if (filesToRestore.Count > 0)
+                return;
+
+            foreach (var (name, state) in subscription)
+            {
+                var command = new PutSubscriptionCommand(databaseName, state.Query, state.MentorNode, RaftIdGenerator.DontCareId)
+                {
+                    Disabled = state.Disabled,
+                    InitialChangeVector = state.ChangeVectorForNextBatchStartingPoint,
+                };
+
+                await database.ServerStore.SendToLeaderAsync(command);
```

Comment:
Add a comment about why we don't wait here for the command to be executed

Reply:
Done